sentry_streams/tests/test_runner.py

```diff
@@ -40,6 +47,16 @@ def create_pipeline() -> Pipeline:
         ctx=test_pipeline,
         function=lambda x: x,
     )
+    _ = Router(
+        name="step5",
+        ctx=test_pipeline,
+        inputs=[map],
+        routing_table={
+            "step6": Branch(name="step6", ctx=test_pipeline),
+            "step7": Branch(name="step7", ctx=test_pipeline),
+        },
+        routing_function=lambda x: "branch1",
+    )
```

Comment:
The MagicMock adapter is not sufficient anymore for this test as you need to test you are returning the right stream for each branch.
Please modify the DummyAdapter to:
- return something as the stream (the name of the step is enough)
- record all the function calls in a list so that you can run assertions on them
- Use that in the test and assert that the adapter methods are called passing the right stream object and the right parameters.